ts/src/okx.ts

```diff
@@ -2291,40 +2291,87 @@ export default class okx extends Exchange {
          * @returns {object} an [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
          */
         if (symbol === undefined) {
             throw new ArgumentsRequired (this.id + ' editOrder() requires a symbol argument');
         }
         await this.loadMarkets ();
         const market = this.market (symbol);
-        if (!market['spot']) {
-            throw new NotSupported (this.id + ' editOrder() does not support ' + market['type'] + ' orders, only spot orders are accepted');
-        }
         const request = {
             'instId': market['id'],
         };
-        const clientOrderId = this.safeString2 (params, 'clOrdId', 'clientOrderId');
-        if (clientOrderId !== undefined) {
-            request['clOrdId'] = clientOrderId;
+        const isStop = this.safeValue (params, 'stop');
+        const tpTriggerPx = this.safeValue2 (params, 'takeProfitPrice', 'tpTriggerPx');
+        const tpOrdPx = this.safeValue (params, 'tpOrdPx', price);
+        const tpTriggerPxType = this.safeString (params, 'tpTriggerPxType');
+        const slTriggerPx = this.safeValue2 (params, 'stopLossPrice', 'slTriggerPx');
+        const slOrdPx = this.safeValue (params, 'slOrdPx', price);
+        const slTriggerPxType = this.safeString (params, 'slTriggerPxType');
+        const conditional = (tpTriggerPx !== undefined) || (slTriggerPx !== undefined) || (type === 'conditional');
+        const clientOrderId = this.safeStringN (params, [ 'clOrdId', 'clientOrderId', 'algoClOrdId' ]);
+        let method = undefined;
+        // Support stop order only
+        if (conditional || isStop) {
+            // Only applicable to Futures and Perpetual swap.
+            if ((!market['future']) && !market['swap']) {
+                throw new NotSupported (this.id + ' editOrder() does not support ' + market['type'] + ' orders, only future & swap orders are accepted');
+            }
+            method = 'privatePostTradeAmendAlgos';
+            request['algoClOrdId'] = clientOrderId;
+            request['algoId'] = id;
+            if (tpTriggerPx !== undefined) {
+                request['newTpTriggerPx'] = this.priceToPrecision (symbol, tpTriggerPx);
+            }
+            if (tpOrdPx !== undefined) {
+                request['newTpOrdPx'] = this.priceToPrecision (symbol, tpOrdPx);
+            }
+            if (tpTriggerPxType !== undefined) {
+                request['newTpTriggerPxType'] = tpTriggerPxType;
+            }
+            if (slTriggerPx !== undefined) {
+                request['newSlTriggerPx'] = this.priceToPrecision (symbol, slTriggerPx);
+            }
+            if (slOrdPx !== undefined) {
+                request['newSlOrdPx'] = this.priceToPrecision (symbol, slOrdPx);
+            }
+            if (slTriggerPxType !== undefined) {
+                request['newSlTriggerPxType'] = slTriggerPxType;
+            }
         } else {
-            request['ordId'] = id;
+            if (!market['spot']) {
```

Comment:
@sc0Vu I just tested and it seems feasible to edit swap orders
![image](https://github.com/ccxt/ccxt/assets/43336371/eb5861a8-69b9-47fa-bcb7-1a7303fdee51)